core/storage/storage_check.go

```diff
@@ -9,24 +9,41 @@ import (
 	"github.com/rancher/agent/utilities/constants"
 	"github.com/rancher/agent/utilities/utils"
 	"golang.org/x/net/context"
+	"path/filepath"
+)
+
+const (
+	rancherSockDir = "/var/run/rancher/storage"
 )
 
 func IsVolumeActive(volume model.Volume, storagePool model.StoragePool, dockerClient *client.Client) (bool, error) {
 	if !isManagedVolume(volume) {
 		return true, nil
 	}
-	vol, err := dockerClient.VolumeInspect(context.Background(), volume.Name)
```

Comment:
This code (and below) and what Darren mentioned before can be put in one separate commit.